functions/src/lobby.ts

```diff
@@ -214,6 +226,59 @@ export const leaveLobby = functions.https.onCall((data: unknown, context): Promi
   });
 });
 
+export const applyLobbySettings = functions.https.onCall(async (data: unknown, context): Promise<void> => {
+  const auth = context.auth;
+  // no auth then you shouldn't be here
+  if (auth === undefined) {
+    throw new functions.https.HttpsError("permission-denied", "Not Signed In");
+  }
+  // validate code
+  if (!isLobbySettingsRequest(data)) {
+    throw new functions.https.HttpsError("invalid-argument", "Invalid lobby code!");
+  }
+
+  const settings = data.lobbySettings;
+
+  if (settings.catfishAmount < 1 || settings.catfishAmount > 3) {
+    throw new functions.https.HttpsError("permission-denied", "Cannot have less than 1 or more than 3 catfish!");
+  }
+
+  if (
+    settings.promptTime < GAME_STATE_DURATIONS_MIN.PROMPT ||
+    settings.promptTime > GAME_STATE_DURATIONS_MAX.PROMPT ||
+    settings.chatTime < GAME_STATE_DURATIONS_MIN.CHAT ||
+    settings.chatTime > GAME_STATE_DURATIONS_MAX.CHAT ||
+    settings.voteTime < GAME_STATE_DURATIONS_MIN.VOTE ||
+    settings.voteTime > GAME_STATE_DURATIONS_MAX.VOTE
+  ) {
+    throw new functions.https.HttpsError("permission-denied", "Timer must be within the proper range!");
+  }
+
+  await db.runTransaction(async (transaction) => {
+    // get lobby doc
+    const lobby = lobbyCollection.doc(data.code);
+    const lobbyInfo = await transaction.get(lobby);
+
+    if (lobbyInfo.exists === false) {
+      throw new functions.https.HttpsError("not-found", "Lobby doesn't exist!");
+    }
+    // check if the request is coming from the host of the game
+    const { uids } = lobbyInfo.data() as Lobby;
+    if (auth.uid !== uids[0]) {
+      throw new functions.https.HttpsError("permission-denied", "Not the host of the game!");
+    }
+
+    transaction.update(lobby, {
+      lobbySettings: {
+        catfishAmount: data.lobbySettings.catfishAmount,
+        promptTime: data.lobbySettings.promptTime,
+        chatTime: data.lobbySettings.chatTime,
+        voteTime: data.lobbySettings.voteTime,
```

Comment:
you created a variable up top called settings so you didnt have to do this long typing